File: ant-sdk/antd-py/examples/08_grpc.py
"""Example 08: Talk to the antd daemon over gRPC instead of REST.

The daemon listens for gRPC on a separate port (default :50051). The shape
mirrors the REST examples — health, then a chunk round-trip — but routed
through ``AntdClient(transport="grpc")`` instead of the default REST client.

Prerequisite: antd daemon running locally with gRPC enabled.
"""

from antd import AntdClient

client = AntdClient(transport="grpc")
print("Connected via gRPC")

status = client.health()
print(f"Daemon healthy: {status.ok}")
print(f"Network: {status.network}")

if not status.ok:
    print("ERROR: antd daemon is not healthy")
    raise SystemExit(1)

raw_data = b"Raw chunk content stored over gRPC"
result = client.chunk_put(raw_data)
print(f"Chunk stored at: {result.address}")

retrieved = client.chunk_get(result.address)
assert retrieved == raw_data, "Chunk round-trip mismatch over gRPC!"
print(f"Retrieved {len(retrieved)} bytes — gRPC round-trip OK!")

client.close()

File: ant-sdk/antd-rust/examples/08-grpc.rs
use antd_client::GrpcClient;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let client = GrpcClient::auto_discover().await?;

    let health = client.health().await?;
    println!("Daemon healthy: {}", health.ok);
    println!("Network: {}", health.network);

    if !health.ok {
        eprintln!("ERROR: antd daemon is not healthy");
        std::process::exit(1);
    }

    let payload = b"Raw chunk content stored over gRPC";
    let put = client.chunk_put(payload).await?;
    println!("Chunk stored at: {}", put.address);

    let retrieved = client.chunk_get(&put.address).await?;
    assert_eq!(
        retrieved.as_slice(),
        payload,
        "chunk round-trip mismatch over gRPC"
    );
    println!("Retrieved {} bytes — gRPC round-trip OK!", retrieved.len());

    Ok(())
}
